pkg/audit/audit_cache_lister.go

```diff
@@ -25,14 +24,19 @@ type CacheLister struct {
 	// Caution: only to be read from while watched is locked, such as through
 	// DoForEach.
 	auditCache client.Reader
-	// watched is the set of objects watched by the audit cache.
-	watched *watch.Set
+	// lister is a delegate like cachemanager that we can use to query a watched set of GKVs.
+	lister Delegate
+}
+
+// wraps DoForEach from a watch.Set.
+type Delegate interface {
```

Comment:
`Delegate` doesn't add meaning here (see [definition](https://www.merriam-webster.com/dictionary/delegate)). In this case, cache manager is delegating to the watch set, but the fact that it is doing so doesn't really tell us why or what that means.

`WatchIterator`?